File: app/src/main/java/br/com/fpgaiad/cleanmvp/presenter/TaskPresenterImpl.java
package br.com.fpgaiad.cleanmvp.presenter;

import br.com.fpgaiad.cleanmvp.model.TasksRepository;
import br.com.fpgaiad.cleanmvp.view.TasksView;

public class TaskPresenterImpl implements TaskPresenter {

    public TasksView view;
    private TasksRepository repository;

    public TaskPresenterImpl(TasksView view, TasksRepository repository) {
        this.view = view;
        this.repository = repository;
    }

    @Override
    public void onAddTaskClicked(String task) {
        if (task.equals("")) {
            view.showMessage("Sorry! Nothing to add here");
        } else {
            view.cleanTaskInput();
            view.showMessage("\"" + task + "\"\nADDED");
            repository.addTask(task);
            view.showTaskList(repository.getTasks());
        }
    }

    @Override
    public void onDeleteTaskClicked(String task) {
        repository.deleteTask(task);
        view.showMessage("DELETED");
        view.showTaskList(repository.getTasks());
    }
}

File: app/src/main/java/br/com/fpgaiad/cleanmvp/presenter/TaskPresenterImpl.kt
package br.com.fpgaiad.cleanmvp.presenter


import br.com.fpgaiad.cleanmvp.model.TasksRepository
import br.com.fpgaiad.cleanmvp.view.TasksView

class TaskPresenterImpl(val view: TasksView?, val repository: TasksRepository) : TaskPresenter {

    override fun onAddTaskClicked(task: String?) {
        if (task.equals("")) {
            view?.showMessage("Sorry, nothing to add here!")
        } else {
            repository.addTask(task)
            view?.let {
                it.cleanTaskInput()
                it.showMessage("\"$task\"\nADDED")
                it.showTaskList(repository.tasks)
            }
        }
    }

    override fun onDeleteTaskClicked(task: String?) {
        task?.let { repository.deleteTask(it) }
        view?.let {
            it.showMessage("DELETED")
            it.showTaskList(repository.tasks)
        }
    }
}